src/Compilers/CSharp/Portable/Symbols/Tuples/TupleFieldSymbol.cs

```diff
@@ -300,5 +359,20 @@ public override bool IsVirtualTupleField
                 return true;
             }
         }
+
+        public sealed override int GetHashCode()
+        {
+            return GetHashCodeInternal();
+        }
+
+        public sealed override bool Equals(object obj)
+        {
+            if (!(obj is TupleVirtualElementFieldSymbol))
```

Comment:
Same here. I do not think we need a type check.
equality of the field index within same tuple type should be enough.